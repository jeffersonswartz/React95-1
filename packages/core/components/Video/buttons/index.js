import Play from './Play';
import Pause from './Pause';
import Stop from './Stop';

export { Play, Pause, Stop };
