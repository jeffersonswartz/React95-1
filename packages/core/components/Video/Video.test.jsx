import React from 'react';
import { waitRender, fireEvent, act } from '../shared/test/utils';
import Video from './Video';

describe('<Video />', () => {
  describe('Snapshot', () => {
    it('should match snapshot', async () => {
      const { container } = await waitRender(
        <Video src="foo/bar/some_video.mp4" />,
      );
      expect(container).toMatchSnapshot();
    });
  });

  describe('events', () => {
    beforeAll(() => {
      // Get and set video duration
      Object.defineProperty(
        global.window.HTMLMediaElement.prototype,
        'duration',
        {
          writable: true,
        },
      );
      Object.defineProperty(
        global.window.HTMLMediaElement.prototype,
        'currentTime',
        {
          writable: true,
        },
      );
      Object.defineProperty(
        global.window.HTMLElement.prototype,
        'offsetWidth',
        {
          writable: true,
        },
      );
    });

    it('should play, pause, and stop', async () => {
      const playStub = jest
        .spyOn(window.HTMLMediaElement.prototype, 'play')
        .mockImplementation(() => {});
      const pauseStub = jest
        .spyOn(window.HTMLMediaElement.prototype, 'pause')
        .mockImplementation(() => {});

      const { container } = await waitRender(
        <Video src="foo/bar/some_video.mp4" />,
      );

      const video = container.querySelector('video');
      act(() => {
        video.dispatchEvent(new window.Event('loadeddata'));
      });

      const [playBtn, stopBtn] = container.querySelectorAll('button');

      // Play
      fireEvent.click(playBtn);
      // Pause
      fireEvent.click(playBtn);
      // Stop
      fireEvent.click(stopBtn);

      expect(playStub).toHaveBeenCalled();
      // Pause + Stop
      expect(pauseStub).toHaveBeenCalledTimes(2);
    });

    it('should change range when video updates', async () => {
      const { container } = await waitRender(
        <Video src="foo/bar/some_video.mp4" />,
      );

      const video = container.querySelector('video');
      // 60s video
      video.duration = 60;
      video.currentTime = 0;

      const videoRange = container.querySelector('input');

      expect(videoRange.value).toBe('0');

      act(() => {
        video.currentTime = 1;
        video.dispatchEvent(new window.Event('timeupdate'));
      });

      expect(videoRange.value).not.toBe('0');
    });

    it('should update video when range updates', async () => {
      class ExtendableMouseEvent extends MouseEvent {
        constructor(type, values) {
          const { offsetX, ...mouseValues } = values;
          super(type, { ...mouseValues, bubbles: true, cancelable: true });

          Object.assign(this, {
            offsetX: offsetX || 0,
          });
        }
      }

      const { container } = await waitRender(
        <Video src="foo/bar/some_video.mp4" />,
      );

      const video = container.querySelector('video');
      // 60s video
      video.duration = 60;
      video.currentTime = 0;

      const videoRange = container.querySelector('input');
      videoRange.offsetWidth = 100;

      expect(videoRange.value).toBe('0');

      act(() => {
        fireEvent(
          videoRange,
          new ExtendableMouseEvent('click', {
            // forward 10%
            offsetX: 10,
          }),
        );
      });

      expect(video.currentTime).toBeGreaterThan(0);
    });
  });
});
