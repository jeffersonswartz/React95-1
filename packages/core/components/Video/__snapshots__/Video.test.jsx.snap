// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Video /> Snapshot should match snapshot 1`] = `
.c2 {
  display: block;
  background-repeat: no-repeat;
  background-position: center;
  background-size: contain;
  width: 16px;
  height: 16px;
}

.c0 {
  padding: 2px;
  background-color: #c3c7cb;
  box-shadow: inset 0.5px 0.5px 0px 0.5px #ffffff, inset 0 0 0 1px #868a8e, 1px 0px 0 0px #000000, 0px 1px 0 0px #000000, 1px 1px 0 0px #000000;
}

.c3 {
  width: 100%;
  padding: 2px;
  display: none;
}

.c9 {
  background-color: #c3c7cb;
  padding: 7px 20px 5px;
  border: none;
  font-size: 12px;
  box-shadow: inset 1px 1px 0px 1px #ffffff, inset 0 0 0 1px #868a8e,1px 1px 0 0px #000000;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c9:disabled {
  color: #868a8e;
}

.c9:focus {
  outline: 1px dotted #000000;
  outline-offset: -5px;
  box-shadow: inset 1px 1px 0px 1px #ffffff, inset -0.5px -0.5px 0px 1px #868a8e, 1px 1px 0 1px #000000;
}

.c9:active {
  padding: 8px 20px 4px;
  outline: 1px dotted #000000;
  outline-offset: -5px;
  box-shadow: inset 0 0 0 1px #868a8e, 0 0 0 1px #000000;
}

.c9.c9,
.c9:active,
.c9:focus {
  width: 20px;
  height: 20px;
  padding: 7px;
  padding: 4px;
}

.c9.c9 svg,
.c9:active svg,
.c9:focus svg {
  fill: #868a8e;
  border-bottom: 1px solid white;
  border-right: 1px solid white;
}

.c1 {
  height: 18px;
  margin-bottom: 2px;
  color: #ffffff;
  padding: 0 2px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #000e7a;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 2px 0;
}

.c5 {
  background-color: #c3c7cb;
  box-shadow: inset 0.5px 0.5px 0px 0.5px #ffffff, inset 0 0 0 1px #868a8e, 1px 0px 0 0px #000000, 0px 1px 0 0px #000000, 1px 1px 0 0px #000000;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 6px;
  margin-bottom: 4px;
  box-shadow: inset 0px 0px 0px 0px, inset 1px 1px 0px 0px #868a8e, 0.5px 0.5px 0px 0.5px #ffffff;
  background-color: #000000;
  height: 50px;
  color: #ffffff;
}

.c7 {
  font-family: 'React95Video-Numbers';
  text-transform: uppercase;
}

.c4 {
  margin-left: auto;
  margin-right: auto;
  margin-bottom: 4px;
  max-width: 250px;
  background-color: #c3c7cb;
  box-shadow: inset 0.5px 0.5px 0px 0.5px #ffffff, inset 0 0 0 1px #868a8e, 1px 0px 0 0px #000000, 0px 1px 0 0px #000000, 1px 1px 0 0px #000000;
  background-color: transparent;
  box-shadow: none;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 40%;
  background-color: #c3c7cb;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-shadow: inset 0.5px 0.5px 0px 0.5px #ffffff, inset 0 0 0 1px #868a8e, 1px 0px 0 0px #000000, 0px 1px 0 0px #000000, 1px 1px 0 0px #000000;
  background-color: transparent;
  box-shadow: none;
}

.c10 {
  -webkit-appearance: none;
  width: 100%;
}

.c10:hover,
.c10:focus,
.c10:active {
  outline: none;
}

.c10::-webkit-slider-runnable-track {
  width: 100%;
  height: 4px;
  background: #000000;
  cursor: pointer;
  border-top: 1px solid;
  border-top-color: #d2d2d2;
  border-right: 2px solid;
  border-right-color: #ffffff;
  border-bottom: 1px solid;
  border-bottom-color: #e6e6e6;
  border-left: 2px solid;
  border-left-color: #ffffff;
  box-shadow: 0px 1px 0px 0px #ffffff;
}

.c10::-webkit-slider-thumb {
  width: 12px;
  height: 20px;
  background-color: #c3c7cb;
  cursor: pointer;
  margin-top: -8px;
  border-top: 1px solid;
  border-top-color: #ffffff;
  border-right: 1px solid;
  border-right-color: #000000;
  border-bottom: 1px solid;
  border-bottom-color: #000000;
  border-left: 1px solid;
  border-left-color: #ffffff;
  box-shadow: inset 0px -1px 0px #868a8e, inset -1px 0px 0px #868a8e, inset 0px 1px 0px #e6e6e6, inset 1px 0px 0px #e6e6e6;
  -webkit-appearance: none;
}

.c10:focus::-webkit-slider-runnable-track {
  background: #000000;
}

.c10::-moz-range-track {
  width: 100%;
  height: 4px;
  background: #000000;
  cursor: pointer;
  border-top: 1px solid;
  border-top-color: #d2d2d2;
  border-right: 2px solid;
  border-right-color: #ffffff;
  border-bottom: 1px solid;
  border-bottom-color: #e6e6e6;
  border-left: 2px solid;
  border-left-color: #ffffff;
  box-shadow: 0px 1px 0px 0px #ffffff;
}

.c10::-moz-range-thumb {
  width: 12px;
  height: 20px;
  background-color: #c3c7cb;
  cursor: pointer;
  margin-top: -8px;
  border-top: 1px solid;
  border-top-color: #ffffff;
  border-right: 1px solid;
  border-right-color: #000000;
  border-bottom: 1px solid;
  border-bottom-color: #000000;
  border-left: 1px solid;
  border-left-color: #ffffff;
  box-shadow: inset 0px -1px 0px #868a8e, inset -1px 0px 0px #868a8e, inset 0px 1px 0px #e6e6e6, inset 1px 0px 0px #e6e6e6;
  -webkit-appearance: none;
}

.c10::-webkit-slider-thumb {
  height: 18px;
  margin-top: -7px;
  width: 10px;
}

.c10[value='0']::-webkit-slider-thumb {
  margin-left: -2px;
}

<div>
  <div
    class="c0"
    style="width: 260px;"
  >
    <div
      class="c1"
    >
      <i
        class="c2"
        height="16"
        name="mplayer_1_13"
        style="margin-right: 4px; background-image: url(buffer-16-4-1);"
        width="16"
      />
      some_video.mp4
       (Openning)
    </div>
    <video
      class="c3"
    >
      <source
        src="foo/bar/some_video.mp4"
        type="video/mp4"
      />
    </video>
    <div
      class="c4"
    >
      <div
        class="c5"
      >
        <div
          class="c6"
          display="flex"
          width="40%"
        >
          <span
            class="c7"
            style="margin-top: auto;"
          >
            00:00
          </span>
          <span
            class="c7"
            style="height: 12px;"
          >
            Openning
          </span>
        </div>
        <div
          class="c6"
          display="flex"
          width="40%"
        >
          <span
            class="c7"
            style="margin-top: auto; font-size: 22px;"
          >
            00:00
          </span>
          <span
            class="c7"
            style="height: 12px;"
          >
            time
          </span>
        </div>
      </div>
      <div
        class="c8"
      >
        <button
          class="sc-pscky c9"
          disabled=""
        >
          <i
            class="c2"
            height="16"
            name="user_4"
            style="margin-top: -2px; background-image: url(buffer-32-4-1);"
            width="16"
          />
        </button>
        <button
          class="sc-pscky c9"
          disabled=""
        >
          <svg
            height="6"
            width="6"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h306v306H0z"
            />
          </svg>
        </button>
        <input
          class="sc-pAyMl c10"
          max="100"
          min="0"
          step="1"
          style="width: 70%; margin-left: 20px;"
          type="range"
          value="0"
        />
      </div>
    </div>
  </div>
</div>
`;
